src/ManageCourses.Api/Services/AccessRequests/AccessRequestService.cs

```diff
@@ -20,48 +20,55 @@ public AccessRequestService(IManageCoursesDbContext context, IAccessRequestEmail
             _emailService = emailService;
         }
 
+        public DbContext RawContext => (DbContext)_context;
+
         public void LogAccessRequest(AccessRequest request, string requesterEmail)
         {
-            using (var transaction = ((DbContext)_context).Database.BeginTransaction())
+            // Strategy is required for compatibility with EnableRetryOnFailure in Startup. Ref https://docs.microsoft.com/en-gb/azure/architecture/best-practices/retry-service-specific#sql-database-using-entity-framework-core
+            var strategy = RawContext.Database.CreateExecutionStrategy();
+            strategy.Execute(() =>
```

Comment:
Can we not expose it in dbcontext interface?
Rather than this casting? `DbContext RawContext => (DbContext)_context`

Reply:
Yeah I guess so